src/pages/communication-page/cat-a-mod1/communication.cat-a-mod1.page.ts

```diff
@@ -0,0 +1,383 @@
+        select(getUntitledCandidateName),
+      ),
+      candidateDriverNumber$: currentTest$.pipe(
+        select(getJournalData),
+        select(getCandidate),
+        select(getCandidateDriverNumber),
+        map(formatDriverNumber),
+      ),
+      candidateProvidedEmail$: currentTest$.pipe(
+        select(getJournalData),
+        select(getCandidate),
+        select(getCandidateEmailAddress),
+        take(1),
+      ),
+      communicationEmail$: this.store$.pipe(
+        select(getTests),
+        select(getCurrentTest),
+        select(getCommunicationPreference),
+        select(getCommunicationPreferenceUpdatedEmail),
+      ),
+      communicationType$: currentTest$.pipe(
+        select(getCommunicationPreference),
+        select(getCommunicationPreferenceType),
+      ),
+      candidateAddress$: currentTest$.pipe(
+        select(getJournalData),
+        select(getCandidate),
+        select(getPostalAddress),
+      ),
+      conductedLanguage$: currentTest$.pipe(
+        select(getCommunicationPreference),
+        select(getConductedLanguage),
+      ),
+    };
+
+    const {
+      candidateProvidedEmail$,
+      communicationEmail$,
+      communicationType$,
+      conductedLanguage$,
+    } = this.pageState;
+
+    this.merged$ = merge(
+      candidateProvidedEmail$.pipe(map(value => this.candidateProvidedEmail = value)),
+      communicationEmail$.pipe(map(value => this.communicationEmail = value)),
+      communicationType$.pipe(map(value => this.communicationType = value as CommunicationMethod)),
+      conductedLanguage$.pipe(tap(value => configureI18N(value as Language, this.translate))),
+    );
+
+    this.subscription = this.merged$.subscribe();
+
+    if (this.shouldPreselectADefaultValue()) {
+      this.initialiseDefaultSelections();
+    }
+
+    this.restoreRadiosFromState();
+    this.restoreRadioValidators();
+  }
+
+  ionViewWillEnter(): boolean {
+    if (this.subscription.closed && this.merged$) {
+      this.subscription = this.merged$.subscribe();
+    }
+
+    return true;
+  }
+
+  ionViewDidLeave(): void {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
+
+  onSubmit() {
+    Object.keys(this.form.controls).forEach(controlName => this.form.controls[controlName].markAsDirty());
+    if (this.form.valid) {
+      this.deviceAuthenticationProvider.triggerLockScreen()
+        .then(() => {
+          this.store$.dispatch(new CommunicationSubmitInfo());
+          this.navController.push(CAT_AMOD1.WAITING_ROOM_TO_CAR_PAGE)
+            .then(() => {
+              const waitingRoomPage =
+              this.navController.getViews().find(view => view.id === CAT_AMOD1.WAITING_ROOM_PAGE);
+              if (waitingRoomPage) {
+                this.navController.removeView(waitingRoomPage);
+              }
+              const communicationPage =
+                this.navController.getViews().find(view => view.id === CAT_AMOD1.COMMUNICATION_PAGE);
```

Comment:
I'm just checking because I don't know myself, is `CAT_AMOD1` agreed for this, as opposed to `CAT_A_MOD_1`?

Reply:
Ignore this, Szabi confirmed it's appropriate.